packages/lodestar/src/util/queue.ts

```diff
@@ -0,0 +1,78 @@
+import {EventEmitter} from "events";
+
+export enum QueueErrorCode {
+  ERR_QUEUE_ABORTED = "ERR_QUEUE_ABORTED",
+  ERR_QUEUE_THROTTLED = "ERR_QUEUE_THROTTLED",
+}
+
+export class QueueError extends Error {
+  code: QueueErrorCode;
+  constructor(code: QueueErrorCode, msg?: string) {
+    super(msg || code);
+    this.code = code;
+  }
+}
+
+/**
+ * EventEmitter-based job queue
+ */
+export class JobQueue extends EventEmitter {
```

Comment:
This queue implementation seems weird and overly comlicated. Maybe use async generators like here:
https://github.com/ChainSafe/lodestar/pull/1551/files#diff-009319ca2922f4fbfff417a57221eb69R14